src/auth/src/credentials/service_account.rs

```diff
@@ -377,15 +377,29 @@ impl TokenProvider for ServiceAccountTokenProvider {
 }
 
 #[derive(Default, Clone)]
-struct ServiceAccountTokenGenerator {
+pub(crate) struct ServiceAccountTokenGenerator {
     service_account_key: ServiceAccountKey,
     audience: Option<String>,
     scopes: Option<String>,
     target_audience: Option<String>,
 }
 
 impl ServiceAccountTokenGenerator {
-    fn generate(&self) -> Result<String> {
+    #[cfg(google_cloud_unstable_id_token)]
+    pub(crate) fn new_id_token_generator(
```

Comment:
This doesn't generate an id token. It generates an assertion that can be exchanged for an id token.

Consider updating the name to something like `new_assertion_generator`